app.go

```diff
@@ -224,3 +226,75 @@ func executeQuote(cmd command) bool {
 	// send the quote to the user
 	return true
 }
+
+func executeBuy(cmd command) bool {
+	//Gotta check users money and add a reserved portion
+	account := accountStore.GetAccount(cmd.UserID)
+
+	if account == nil {
+		log.Noticef("User %s does not have an account", account)
+	}
+
+	stockSymbol := cmd.Args[0]
+	dollarAmount, err := currency.NewFromString(cmd.Args[1])
+	//dollarAmount, err := strconv.ParseFloat(cmd.Args[1], 64)
+	if err != nil {
+		log.Noticef("Dollar amount %s is invalid", cmd.Args[1])
+		return false
+	}
+	//User wants to buy y worth of x shares.
+	userQuote, err := quotecache.GetQuote(cmd.UserID, stockSymbol)
+
+	if err != nil {
+		log.Noticef("Quote of stock %s for user %s is invalid", stockSymbol, cmd.UserID)
+		return false
+	}
+
+	wholeShares, cashRemainder := userQuote.Price.FitsInto(dollarAmount)
+
+	if wholeShares == 0 {
+		log.Notice("Amount specified to buy less than single stock unit")
+		return true
+	}
+
+	dollarAmount.Sub(cashRemainder)
+	account.RemoveFunds(dollarAmount)
+
+	return account.AddToBuyQueue(stockSymbol, wholeShares)
+}
+
+func executeSell(cmd command) bool {
+	account := accountStore.GetAccount(cmd.UserID)
+
+	if account == nil {
+		log.Noticef("User %s does not have an account", account)
+	}
+
+	stockSymbol := cmd.Args[0]
+	dollarAmount, err := currency.NewFromString(cmd.Args[1])
+
+	if err != nil {
+		log.Noticef("Dollar amount %s is invalid", cmd.Args[1])
+		return false
+	}
+
+	userQuote, err := quotecache.GetQuote(cmd.UserID, stockSymbol)
+
+	if err != nil {
+		log.Noticef("Quote of stock %s for user %s is invalid", stockSymbol, cmd.UserID)
+		return false
+	}
+
+	wholeShares, cashRemainder := userQuote.Price.FitsInto(dollarAmount)
+
+	if wholeShares == 0 {
+		log.Notice("Amount specified to sell less than single stock unit")
+		return true
+	}
+
+	dollarAmount.Add(cashRemainder)
```

Comment:
Confirmed. Difference between before and after balances in this screenshot it $754.77 but it should be $529.03. :
<img width="624" alt="screen shot 2017-01-19 at 11 32 19 pm" src="https://cloud.githubusercontent.com/assets/6734837/22140190/a4cef846-de9f-11e6-81ab-6a708e65de35.png">